### PR TITLE
Slim down 0.15

### PR DESCRIPTION
**Pull request by hmaarrfk**
@jni 
```
skip: True
```

is probably the more stable way of skipping 2.7. I haven't see `six` in the codebase. I'm guessing that is an artifact of time ago.

Are we good with removing dask from the default dependencies? What is the consensus on that? 

Checklist
* [ ] Used a fork of the feedstock to propose changes
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by jni** [comment]
@hmaarrfk, of the scikit-image maintainers, I think only you and I really care about the conda build. I personally like conda to have all the optional dependencies, since I tend to use conda packages as a more "batteries included" thing than PyPI ones. I'm not strongly opposed to removing them though.

Maybe @soupault or @jakirkham have comments? (We are talking about removing the dependency on dask from the scikit-image conda-forge package, since it is now an optional dependency.)

Thanks for the maintenance @hmaarrfk!

**Comment by hmaarrfk** [comment]
In that case maybe creating a core package is better. Dask is just not so much a light dependency in conda.

**Comment by jni** [comment]
@hmaarrfk very good idea about creating a core package — I think it's the right solution — but currently it's broken!

I also don't think we should update the readme, except to add a section that scikit-image-core is *also* produced. I think most users will want plain scikit-image.

**Comment by jni** [comment]
Also, could you fiddle with your editor settings? A lot of the files are missing the final newline, see the diff.

**Comment by hmaarrfk** [comment]
> Also, could you fiddle with your editor settings? A lot of the files are missing the final newline, see the diff.

I don't write any of these files, they are autogenerated. I only wrote the meta.yaml, and when I added the new line, the linter complained with a non-sensical message.

**Comment by jni** [comment]
Weird. Maybe they made a change to conda-smithy...

**Comment by soupault** [comment]
>of the scikit-image maintainers, I think only you and I really care about the conda build

Sorry, I never really had time to properly investigate the conda-forge infrastructure around skimage. I _do_ care about conda releases being stable and up-to-date, though!

>I personally like conda to have all the optional dependencies, since I tend to use conda packages as a more "batteries included" thing than PyPI ones. I'm not strongly opposed to removing them though.

Same for me, more or less. In most of the projects I have mixed conda+pip environment, so I usually stick to installing skimage from pip (always the latest version, more convenient switching between the release and the development versions). It is hard to say how many users are running conda on embedded systems (where the minimal set of dependencies is preferred), but, I think, there are some. So I'd vote for harmonizing the dependencies between pip and conda distributions (i.e. dropping the dependency on dask here).

Having [core] and [full] packages would be the best option, obviously.

**Comment by jni** [comment]
@soupault unfortunately conda doesn't do package variants like pip does. But different packages work!

File: README.md [hunk 1]
```diff
@@ -112,20 +91,21 @@ Current release info
...
 
-Installing `scikit-image` from the `conda-forge` channel can be achieved by adding `conda-forge` to your channels with:
+Installing `scikit-image-core` from the `conda-forge` channel can be achieved by adding `conda-forge` to your channels with:
 
 ```
 conda config --add channels conda-forge
 ```
 
-Once the `conda-forge` channel has been enabled, `scikit-image` can be installed with:
+Once the `conda-forge` channel has been enabled, `scikit-image, scikit-image-core` can be installed with:
```
Comment (jni) [inline README.md]: @hmaarrfk so you're saying that this is auto-generated? That's probably not desirable in this case... =\

Comment (hmaarrfk) [reply README.md]: Yeah, this is auto generated.

I would consider conda-forge more bleeding edge. Anaconda Defaults regularly syncs up with conda-forge so the work we do here eventually gets upstreamed to there..

**Comment by hmaarrfk** [comment]
@soupault I've read by some of the creators of conda-forge taht the idea of conda-forge was to break away from the name of the package meaning "the minimum requirements" allowing users to "install and go" with a "battery included" approach.

This was specifically the design of conda-forge.

Now, they let us, the package maintainers, take control over specific packages, but I think having a "core" package that allows maintainers to eventually slim down their package is the right move.

File: README.md (hunk 1 above)
Comment (hmaarrfk) [reply README.md]: I also do not think that anybody is going on this page to learn how to use "scikit-image", just the packages. scikit-image's docs are still valid, and shouldn't mention core. Packages should be able to read this "meta.yaml" file and understand what is going on.

**Comment by jni** [comment]
@hmaarrfk maybe close this now?

**Comment by jni** [comment]
Incidentally, a potential downside of using github rather than pypi as the release source: the new version bot is nowhere to be seen. I wonder if it only works when the source code comes from PyPI.

**Comment by jakirkham** [comment]
The bot is run as a cron job. So there can be some delay.

**Comment by hmaarrfk** [comment]
Anyway, with matplotlib core this PR isn't so useful.

**Comment by jni** [comment]
@jakirkham Ah! The last time it updated basically a second after we published to PyPI, coincidentally it turns out, so @NelleV and I were super impressed! 😅

**Comment by jakirkham** [comment]
Yeah it's a cron job that happens pretty frequently. Though I don't recall the exact cadence.

We've talked about making it a web service ( https://github.com/regro/cf-scripts/issues/54 ), but that probably requires a champion. 😉

**Pull request closed without merging**